src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,148 @@
+
+## Keycloak Installation
+### By Docker
+Make sure [Docker Engine](https://docs.docker.com/engine/install/) and [Docker Compose](https://docs.docker.com/compose/install/) has been installed in your system.
+Two different ways are available to install Keycloak using docker as mentioned below.
+
+1. Keycloak using docker image
+Run the following docker command to serve the Keycloak image. Keycloak will be available on `http://localhost:8080.
+```bash
+docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:24.0.2 start-dev
+ ```
+
+2. Keycloak with postgres database using docker-compose
+```
+services:
+  postgres:
+    image: postgres:alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: keycloak
+      POSTGRES_PASSWORD: keycloak
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:24.0.1
+    command: ["start-dev"]
+    environment:
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    ports:
+      - 8001:8080 # map keycloak port 8080 to 8001
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save the above code in `compose.yaml` file and run `docker compose up` in a terminal.
+
+Docker compose serves Keycloak on `http://localhost:8001`.
+
+### By Distribution File
+There is another way to set up Keycloak using the Keycloak distribution file.
+1. Download the Keycloak distribution
+    ```
+    export KC_VERSION=24.0.4
+    curl -LO  https://github.com/keycloak/keycloak/releases/download/"${KC_VERSION}"/keycloak-"${KC_VERSION}".zip
+    ```
+
+2. Unzip the Keycloak distribution package
+    ```
+    unzip keycloak-${KC_VERSION}.zip
+    ```
+
+3. Install openjdk
+    ```
+    sudo apt install openjdk-17-jdk
+    ```
+
+4. Navigate to the Keycloak directory
+    ```
+    cd keycloak-${KC_VERSION}
+   ```
+
+5. When we start the server for the first time, we have to set the admin user and the admin password:
+    ```
+   KEYCLOAK_ADMIN=admin KEYCLOAK_ADMIN_PASSWORD=admin ./bin/kc.sh start-dev
+   ```
+    > Note: if port 8080 is already used, then you can map the Keycloak port using the Keycloak environment variable 'KC_HTTP_PORT=8001'
+
+6. When we start again, it is not necessary to set these variables again. You can start the server with:
+    ```
+   ./bin/kc.sh start-dev
+   ```
+   > Note: start-dev runs the Keycloak application in DEV-mode. Do not use this for production.
+
+By default, the Keycloak server is on port 8080 for http and 8084 for https. They are only served from the localhost loopback address 127.0.0.1:
+Keycloak serves on `http://localhost:<port>`.
+
+## Why Keycloak Matters?
```

Comment:
```suggestion
## Why Keycloak Matters
```

A phrase like this is a "statement" and not actually a question, even though it starts with "Why"